src/main/java/core/basesyntax/strategy/impl/PurchaseStrategyOperationImpl.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class PurchaseStrategyOperationImpl implements OperationHandler {
+
+    @Override
+    public void handle(FruitTransaction transaction) {
+        String fruitName = transaction.getFruit();
+        int purchaseQuantity = transaction.getQuantity();
+        int oldQuantity = Storage.fruits.get(fruitName);
+        Storage.fruits.put(fruitName, oldQuantity - purchaseQuantity);
```

Comment:
what happens, if I try to buy more fruit than you have in store? May the store contain a negative amount of fruit?